app/src/main/kotlin/com/mapzen/erasermap/view/MainActivity.kt

```diff
@@ -145,15 +149,34 @@ public class MainActivity : AppCompatActivity(), MainViewController, Router.Call
     private fun getApiKeys() {
         var dl: ManifestDownLoader = ManifestDownLoader()
         dl.download(apiKeys, {
-            if(apiKeys?.getValhallaApiKey()== null) {
+            if (apiKeys?.getValhallaApiKey() == null) {
                 apiKeys?.setValhallaApiKey(BuildConfig.VALHALLA_API_KEY)
             }
-            if(apiKeys?.getVectorTileApiKeyReleaseProp()== null) {
+            if (apiKeys?.getVectorTileApiKeyReleaseProp() == null) {
                 apiKeys?.setVectorTileApiKeyReleaseProp(BuildConfig.VECTOR_TILE_API_KEY)
             }
+            if(apiKeys?.getMinVersion() != null) {
+                checkIfUpdateNeeded()
+            }
         })
     }
 
+    public fun checkIfUpdateNeeded() {
+        var currentVersion: Int = getPackageManager().getPackageInfo(getPackageName(), 0).versionCode;
```

Comment:
With Gradle we can just use `BuildConfig.VERSION_CODE` instead of package manager.